content/en/docs/usage/certificate.md

```diff
@@ -91,71 +90,159 @@ The `Certificate` will be issued using the issuer named `ca-issuer` in the
+
+Setting the `rotationPolicy: Always` or change the Certificate's spec won't
+rotate the private key immediately. In order to get the private key secret
+rotated, the certificate objects must be reissued. A certificate object is
+reissued with either:
+
+- when the X.509 certificate is nearing expiry,
+- when a reissuance is manually triggered with the following:
+  ```sh
+  kubectl cert-manager renew cert-1
+  ```
+  Note that the above command requires the [kubectl
+  cert-manager](/docs/usage/kubectl-plugin/#renew) plugin.
+
+{{% pageinfo color="warning" %}}
+
+**❌** Deleting the Secret resource associated with a Certificate resource is
+**not a recommended solution** for manually rotating the private key. The
+recommended way to manually rotate the private key is to trigger the reissuance
+of the Certificate resource with the following command (requires the [`kubectl
+cert-manager`](/docs/usage/kubectl-plugin/#renew) plugin):
+
+```sh
+kubectl cert-manager renew cert-1
 ```
 
-There are two supported rotation policies:
+{{% /pageinfo %}}
+
+### The `rotationPolicy` setting
+
+The possible values for `rotationPolicy` are:
+
+| Value                  | Description                                                   |
+| ---------------------- | ------------------------------------------------------------- |
+| `Never` (default)      | cert-manager reuses the existing private key on each issuance |
+| `Always` (recommended) | cert-manager regenerates a new private key on each issuance   |
+
+With `rotationPolicy: Never`, a private key is only generated if one does not
+already exist in the target Secret resource (using the `tls.key` key). All
+further issuances will re-use this private key. This is the default in order to
+maintain compatibility with previous releases.
+
+With `rotationPolicy: Always`, a new private key will be generated each time an
+action triggers the reissuance of the certificate object (see
+[actions](#actions-triggering-private-key-rotation) above). Note that if the
+private key secret already exists when creating the certificate object, the
+existing private key will not be used, since the rotation mechanism also
+includes the initial issuance.
+
+{{% pageinfo color="info" %}}
 
-* **Never** (default): a private key is only generated if one does not already exist in
-  the target Secret resource (using the `tls.key` key). All further issuance's will re-use
-  this private key. This is the default in order to maintain compatibility with previous releases.
-* **Always**: a new private key will be generated each time a new certificate is issued.
-  It is recommended you configure this `rotationPolicy` on your Certificate resources as it
-  is good practice to rotate private keys when a certificate is renewed.
+👉 It is recommended you configure `rotationPolicy: Always` on your certificate
+resources. It is considered to be a good practice to rotate private keys when a
+certificate is renewed.
```

Comment:
```suggestion
👉 We recommend that you configure `rotationPolicy: Always` on your certificate
resources. It is considered to be a good practice to rotate private keys when a
certificate is renewed.
```

Can we link to a reference backing up this claim good practice?

Reply:
I cannot find any good reference to support this claim. By reading some discussions:

- https://community.letsencrypt.org/t/should-i-always-generate-a-new-csr-private-key/100985
- https://security.stackexchange.com/questions/27810/should-i-change-the-private-key-when-renewing-a-certificate

People seem to suggest that:
1. RSA/ECDSA private keys can't (mostly) be guessed, which means there is less of a need for renewing them (unlike passwords that are way shorter),
2. But at the same time, an RSA/ECDSA private keys may be leaked through some old backup, leading to a possible man-in-the-middle situation.

Looking around, I found that:

- certbot renews the private key while renewing the cert by default, but there is nothing explaining why;
- NIST's [Guidelines for the Selection, Configuration, and Use of Transport Layer Security (TLS) Implementations](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-52r2.pdf) does not indicate when to renew the private key; the only bit that talks about what to do with a server private key is this:
  > The security of the server’s private key is critical to the security of TLS. If the server’s private key is weak or can be obtained by a third party, the third party can masquerade as the server to
all clients.
- NIST's [Recommendation for Key Management](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-57pt1r5.pdf) has a nice table but no claim on when the private key should be renewweed:
  ![image](https://user-images.githubusercontent.com/2195781/127500092-d779df24-8c3c-445d-a092-4e119757ce83.png)

- f5-bigip suggests to renew the private key (https://support.f5.com/csp/article/K14620):

  > Note: When renewing an SSL certificate from a CA, F5 recommends that you generate a new certificate signing request (CSR) and private key. Although some CAs allow you to renew a certificate by using the existing CSR on file, this method is less secure as it retains the existing private key. To generate a new CSR, follow the procedure for Creating an SSL CSR and private key.


@SgtCoDFish Do you know a place where that would be written? Or should I rather remove this claim and say that "the cert-manager team suggests users renewing the private key along with the certificate"?

Reply:
I think one big reason why you _would_ want to, is that if you were to renew a certificate using the same private key, and then after the initial certificates expiry learn that the private key had actually been leaked, the new certificate would be valid to use with the old private key, thus the 'rotation' didn't really provide any more protection and really just extended the lifetime of the existing certificate.

Reply:
James is spot on. I don't have like an authoritative source to hand for why you'd want to rotate as much as possible, but certainly I think it's best practice. 

The longer a key is used, the more likely it is to be exposed. Keys should be renewed "as often as possible" for me; not only for security reasons but also operational reasons to show that it can be done, so that if it has to be done in an emergency the org has confidence that they can carry it out.

Reply:
I tried my best to formulate a paragraph that "justifies" our point of view. Please forgive me if any of this is untrue. 🔥

👉 We recommend that you configure `rotationPolicy: Always` on your Certificate resources. 

Rotating both the certificate and the private key simultaneously prevents the risk of issuing a certificate associated with an exposed private key.

Another benefit to renewing the private key regularly is to let you be confident that the private key rotation can be done in case of an emergency.

More generally, it is a good practice to be rotating the keys as often as possible, reducing the risk associated with compromised keys.